src/silx/app/compare/CompareImagesWindow.py

```diff
@@ -138,7 +138,7 @@ def readData(self, urlPath: str):
 
         if data.ndim == 2:
             return data
-        if data.ndim == 3 and data.shape[0] in set(3, 4):
+        if data.ndim == 3 and data.shape[0] in set([3, 4]):
```

Comment:
RGB(A) channels are usually the last dimension, not the first, to me this should be:

```suggestion
        if data.ndim == 3 and data.shape[2] in set([3, 4]):
```

Reply:
Done